The PageObjeckt module has been created:

scokpius opened the pull request:
- classes are described in accordance with the PageObjeckt class diagram
- all attributes and methods for working with them are described

scokpius committed:
The PageObjeckt module has been created:
- classes are described in accordance with the PageObjeckt class diagram
- all attributes and methods for working with them are described

vladimir-moshkov commented on Task_PageObject_the_OOP_model/src/main/java/constant/URL.java:
```diff
@@ -0,0 +1,7 @@
+
+public class URL {
+    public static final String HOTWIRE = "https://www.hotwire.com/";
+    public static final String HOTWIRE_HOTELS = "https://www.hotwire.com/";
```
Константы одинаковые. Подумай, тебе точно нужна HOTWIRE_HOTELS ?

vladimir-moshkov commented on Task_PageObject_the_OOP_model/src/main/java/constant/XPathCSS.java:
```diff
@@ -0,0 +1,40 @@
+public class XPathCSS {
+    public static final String BUTTON_CURRENCY_XPATH = "//*[@id=\"currency\"]";
+    public static final String BUTTON_CURRENCY_CONTAINER_XPATH = "//*[@id=\"currency--container\"]";
+    public static final String BUTTON_CURRENCY_ELEMENTS_XPATH = "//*[@id=\"currency--container\"]/li[4]";
```
никогда не используй такого рода xpath (li[4], div[5] и т.д.)
Если верстка минимально поменяется, у тебя всё съедет и поломается
Пример плохого xpath: 
//*[@id="root"]//form/div[2]/div[2]/div/div/div/div[1]
Пример хорошего на том же элементе:
//*[@class="hw-form-group form-group floating-label not-empty has-icon is-readonly"]

Пример плохого xpath:
//*[@id=\"currency--container\"]/li[4]
Пример хорошего на том же элементе:
//*[@data-currency-code="CHF"]   -- Здесь вообще можно параметризацию ввести //*[@data-currency-code="%s"] и искать валюту по названию

vladimir-moshkov replied:
/nav/div/ul/li/a - это тоже очень плохой xpath

vladimir-moshkov replied:
Пересмотри их все. Почти все надо переделать

scokpius committed:
Merge pull request #1 from scokpius/oop-model

Description of OOP Modal in code:

scokpius committed:
Merge branch 'master' of https://github.com/scokpius/Practical_Tasks into page_object_modal

scokpius committed:
Refactoring the code:
- created abstract class CalendarDate, Guests, SearchOptions
- creating implementation classes for abstract classes
- moving the locator into a separate class

vladimir-moshkov commented on Task_PageObject_the_OOP_model/src/main/java/constant/XPathCSS.java:
```diff
@@ -0,0 +1,40 @@
+    public static final String HOME_PAGE_BUTTON_GUESTS_DROPDOWN_XPATH = "//*[@class=\"guest-fields\"]";
+    public static final String HOME_PAGE_BUTTON_FIND_XPATH = "//*[@id=\"root\"]//form/div[2]/div[3]/button";
+    public static final String HOME_PAGE_DROPDOWN_CALENDAR_XPATH = "//*[@id=\"tipContent-888\"]";
+    public static final String HOME_PAGE_BUTTON_PREV_MONTH_XPATH = "//*[@role=\"application\"]/div[1]/button";
```
Плохой xpath. 
Можно ориентироваться, например, на дочерний класс с текстом Previous month.
И таких плохих xpath у тебя много

vladimir-moshkov commented on Task_PageObject_the_OOP_model/pom.xml:
```diff
@@ -0,0 +1,61 @@
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
```
Выбери чтото одно,
или junit, или testng

vladimir-moshkov commented on Task_PageObject_the_OOP_model/src/main/java/components/FunctionMenu.java:
```diff
@@ -0,0 +1,23 @@
+
+    public void functionMenuNavigation(String name) {
+        for (int i = 0; i < elementList.size(); i++) {
+            if (elementList.get(i).getText() == name) {
```
Так стринги не сравниваются. У тебя всегда false будет

vladimir-moshkov commented on Task_PageObject_the_OOP_model/src/main/java/components/DropdownMenu.java:
```diff
@@ -0,0 +1,24 @@
+
+    public void selectFromDropdownMenu(String element) {  //
+        $(By.xpath(this.buttonDropdown)).shouldBe(Condition.visible, Duration.ofSeconds(3)).click();
+        $(By.xpath(this.dropdown)).$(By.xpath(element));
```
Что здесь делается с этим элементом? в конце должно быть какое-то действие, я полагаю

vladimir-moshkov commented on Task_PageObject_the_OOP_model/src/main/java/components/guests/GuestsHotels.java:
```diff
@@ -0,0 +1,32 @@
+        super.textAmountRooms = textAmountRoomsHotels;
+        super.textAmountAdult = textAmountAdultHotels;
+        super.textAmountChildren = textAmountChildrenHotels;
+        super.buttonAddRoom = $(String.format(buttonAddAndMinHotels, 1, 2));
```
Посмотри все свои варнинги в проекте и устрани их. 
Здесь ты в String.format засовывваешь строку "//*[@id=\"adaptive-menu\"]/div[2]/div/section/div[2]/button"
что в итоге получится?

scokpius committed:
refactoring the code:
in page transition methods, the returned element is the class of that page,
XPath fix
removing unnecessary dependency
removed the warnings

scokpius committed:
code refactoring
fix XPath, remove getters and setters, fix errors when navigating to pages, create a separate class to change the search for hotel rooms.
